components/bl-treeView-component/src/index.js

```diff
@@ -0,0 +1,139 @@
+import { useState, useEffect, useCallback } from 'react';
+
+const { cn } = BackendlessUI.CSSUtils;
+
+export default function TreeView({ component, eventHandlers }) {
+  const { display, style, classList, data, space } = component;
+  const { onClick } = eventHandlers;
+
+  const [validData, setValidData] = useState([]);
+  const [isOpenItems, setIsOpenItems] = useState([]);
+  const [selectedItemId, setSelectedItemId] = useState('');
+
+  component.closeAll = () => setIsOpenItems(state => state.map(item => ({ ...item, isOpen: false })));
+  component.openAll = () => setIsOpenItems(state => state.map(item => ({ ...item, isOpen: true })));
+
+  const dataValidate = data => {
+    let levelOfNesting = -1;
+
+    const validate = (data) => {
+      levelOfNesting++;
+
+      const validData = data.map(item => {
+        let validItem = { ...item, id: BackendlessUI.UUID.short(), levelOfNesting };
+
+        if (item.children) {
+          validItem = {
+            ...validItem,
+            children: validate(item.children)
+          };
+
+          setIsOpenItems(state => [...state, { id: validItem.id, isOpen: false }]);
+        }
+
+        return validItem;
+      });
+
+      levelOfNesting--;
+
+      return validData;
+    };
+
+    return validate(data);
+  };
+
+  useEffect(() => {
+    if (data) {
+      setValidData(dataValidate(data));
+    }
+  }, [data]);
+
+  const openHandler = useCallback(isOpenItem => {
+    setSelectedItemId(isOpenItem.id);
+    setIsOpenItems(state => state.map(item => item.id === isOpenItem.id ? { ...item, isOpen: !item.isOpen } : item));
```

Comment:
isOpenItem - name which says to us "it's boolean variable", it shouldn't be an object